charts/ingress-nginx/templates/clusterrole.yaml

```diff
@@ -1,4 +1,10 @@
-{{- if and .Values.rbac.create (not .Values.rbac.scope) -}}
+{{- if .Values.rbac.create }}
+
+{{- if and .Values.rbac.scope (not .Values.controller.scope.enabled) -}}
```

Comment:
Why not use the0 `eq` function directly?

Reply:
@tao12345666333 Because this validation error should only trigger when `rbac.scope=true` and `controller.scope.enabled=false`.

Using `eq` would also trigger the validation error if `rbac.scope=false` and `controller.scope.enabled=true`.

Reply:
Is it necessary for us to keep them unequal?

Reply:
Sorry, I don't quite understand your question. 🤔 The valid values of these variables could be found in the description above.

The `controller.scope.enabled` value is also used [in the controller pod](https://github.com/kubernetes/ingress-nginx/blob/0396b88/charts/ingress-nginx/templates/controller-deployment.yaml#L95) where `--watch-namespace` is added. If `controller.scope.enabled=false` and `rbac.scope=true` this causes issues in Kubernetes runtime and this check prevents this misconfiguration.

Reply:
I mean, if we use the same configuration item to control them, is it more direct?

In other words, no longer distinguish between `controller.scope.enabled` and `rbac.scope`, but use the configuration of `global.scope` for control. WDYH?

Reply:
let's put it forward.